django/contrib/admin/views/main.py

```diff
@@ -174,9 +177,25 @@ def get_filters(self, request):
         may_have_duplicates = False
         has_active_filters = False
 
+        lookup_allowed_supports_request = func_supports_parameter(
+            self.model_admin.lookup_allowed, "request"
+        )
+        if not lookup_allowed_supports_request:
+            warnings.warn(
+                f"Add 'request=None' to the signature of "
+                f"{self.model_admin.lookup_allowed.__self__.__class__.__name__}"
+                f".lookup_allowed(). Support  for ModelAdmin subclasses that do not "
+                f"accept this argument is deprecated.",
+                RemovedInDjango60Warning,
+            )
         for key, value_list in lookup_params.items():
             for value in value_list:
-                if not self.model_admin.lookup_allowed(key, value):
+                allowed = (
+                    self.model_admin.lookup_allowed(key, value, request)
+                    if lookup_allowed_supports_request
+                    else self.model_admin.lookup_allowed(key, value)
+                )
+                if not allowed:
```

Comment:
Would it be easier to read with something like the following? 🤔

```suggestion
                params = [key, value]
                if lookup_allowed_supports_request:
                    params += [request]
                if not self.model_admin.lookup_allowed(*params):
```

Reply:
I tried to incorporate a bit of both suggestions :+1: